vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/order_handler.rb

```diff
@@ -17,29 +17,42 @@ def initialize(occupancy)
       def process
         return unless occupancy.account_id?
 
-        ActiveRecord::Base.transaction do
-          create_order
-          create_order_detail
-
-          order.validate_order!
-          order.purchase!
-        end
+        find_or_create_order
+        fulfill_order if occupancy.complete?
```

Comment:
It seems like it might be a good idea to leave the transaction here at this top level. Would we want to roll back any creations if the fulfillment fails? My brain is tired from traveling this morning, so I'm having trouble making arguments for or against.